.github/workflows/e2e.yml

```diff
@@ -0,0 +1,46 @@
+# This is a basic workflow to help you get started with Actions
+
+name: End to end tests
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+        debug_enabled:
+          description: 'Run the build with tmate debugging enabled (https://github.com/marketplace/actions/debugging-with-tmate)'     
+          required: false
+          default: "false"
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      - name: Setup Go environment
+        uses: actions/setup-go@v2.1.3
+        with:
+          # The Go version to download (if necessary) and use. Supports semver spec and ranges.
+          go-version: 1.18.x
+      - name: build Docker images
+        run: make builddockerlocal
+      - name: install kind binaries
+        run: make installkind
+      - name: create kind cluster
+        run: make createkindcluster
+      - name: run end-to-end tests
+        run: make e2elocal
+      - name: Setup tmate session # instructions on https://github.com/marketplace/actions/debugging-with-tmate
```

Comment:
Is there a reason to setup the tmate session in the e2e tests but not in the unit tests?

Reply:
End to end tests use a kind cluster which is easy to investigate using kubectl. Unit tests rely on K8s binaries (etcd, K8s api server) executing, so it’s trickier. We can revisit this later if needed.